.github/workflows/test.yml

```diff
@@ -1,25 +1,74 @@
 name: Build and Test
 on:
   pull_request:
-    paths:
-      # Only run on changes to files that are actually included in the build or this file itself
-      - "code/code/**.cpp"
-      - "code/code/**.cc"
-      - "code/code/**.h"
-      - ".github/workflows/test.yml"
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     name: Build and Test
     steps:
       - name: Checkout this commit
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - run: sudo apt-get update
-      - run: sudo apt-get install software-properties-common
-      - run: sudo apt-get update
-      - run: sudo apt-get install --yes --no-install-recommends build-essential libboost-dev libboost-program-options-dev libboost-regex-dev libboost-filesystem-dev libboost-system-dev libmariadbclient-dev scons libcurl4-openssl-dev
-      - run: scons -C code -j`nproc` -Q pretty=0 debug=1 sanitize=1 olevel=2 sneezy
+      - name: Install dependencies
+        run: |
+          set -e
+          sudo apt-get update
+          export DEBIAN_FRONTEND=noninteractive
+          sudo apt-get install --yes --no-install-recommends g++-9 build-essential libboost-dev libboost-program-options-dev libboost-regex-dev libboost-filesystem-dev libboost-system-dev libmariadb-dev scons libcurl4-openssl-dev ca-certificates mariadb-server
+
+      - name: Verify, configure, and seed a test database
+        run: |
+          set -e
+          sudo systemctl is-active --quiet mariadb || sudo systemctl start mariadb
+          sudo mariadb -e "create database sneezy; create database immortal;"
+          sudo mariadb -e "create user 'runner'@'localhost'; grant all on sneezy.* to 'runner'@'localhost'; grant all on immortal.* to 'runner'@'localhost';"
+          for db in immortal sneezy ; do
+            for phase in tables views data ; do
+              [ -d "_Setup-data/sql_$phase/$db" ] || continue
+              for sql in _Setup-data/sql_$phase/$db/*.sql ; do
+                echo "loading '$sql'"
+                mysql $db < $sql
+              done
+            done
+          done
+
+      - name: Restore cached build objects from master branch
```

Comment:
We now have a sample build that did *not* have the cache, so the next build, which will use the cache, will show us the difference in speed.

Reply:
We won't have a cached build to utilize until the master branch has had changes merged to it at least once, as that's what triggers the workflow that creates the cache.

The reason I didn't have the test.yml workflow create a cache of its build is because `actions/cache` doesn't allow caches to be shared between PRs for some reason. It's just a cache miss every time, no matter what. There's a longstanding issue open about that "feature" in its repo.

But caches are shareable with PRs if the cache was made from the branch the PR is trying to merge with, so therefore if we keep a cached build from the newest version of the master branch, any new PR should be able to basically compile just the "diff" between itself and master.

So merging this PR will trigger the script to create a build cache from the master branch, after which point any new PR or existing PR that's rebased/updated will be able to utilize that cache to speed up its compilation.